pages/data-cleanup.tsx

```diff
@@ -260,6 +273,43 @@ export default function DataCleanupPage(props) {
     dropdownData,
   ]);
 
+  // When formData changes, re-run validity checks
+  useEffect(() => {
+    let allValid = true;
+    if (formData) {
+      if (formData.contactRecord) {
+        if (!contactRecordSchema.isValidSync(formData.contactRecord)) {
+          allValid = false;
+        }
+      }
+      if (formData.emails && allValid) {
+        for (const email of formData.emails) {
+          if (!emailSchema.isValidSync(email)) {
+            allValid = false;
+            break;
+          }
+        }
+      }
```

Comment:
I think this might look better as it's own function.
`
if(formData.emails && allValid){
        allValid = IsEmailsValid(formData.emails)
}`

Reply:
Similar for the rest here